### PR TITLE
add happy path insight dashboards test

### PR DESCRIPTION
**Pull request by unclejustin**
Related to https://github.com/sourcegraph/sourcegraph/issues/24455

Tests the following:

- Redirect to all dashboards page
- Telemetry on the dashboards list page
- Dashboard not found
- View dashboard

### CODE REVIEW

File: client/web/src/enterprise/insights/pages/dashboards/dashboard-page/components/dashboards-content/DashboardsContent.test.tsx [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+        },
+    },
+}))
+
+const mockTelemetryService = {
+    log: sinon.spy(),
+    logViewEvent: sinon.spy(),
+}
+
+const fakeApi = new FakeDefaultCodeInsightsBackend()
```
Comment (vovakulikov) [inline client/web/src/enterprise/insights/pages/dashboards/dashboard-page/components/dashboards-content/DashboardsContent.test.tsx:35]: @unclejustin I think in the case when we test CRUD operations it makes sense to test API handlers as well. In the case of testing the GQL backend we could mock apollo backend and do not mock our abstraction API class here

Comment (unclejustin) [reply client/web/src/enterprise/insights/pages/dashboards/dashboard-page/components/dashboards-content/DashboardsContent.test.tsx:35]: For sure. I had intended to have this done before you saw it but got wildly distracted chasing down a Reach testing bug 😬 .

File: client/web/src/enterprise/insights/pages/dashboards/dashboard-page/components/dashboards-content/DashboardsContent.test.tsx [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+        const chooseDashboard = await waitFor(() => screen.getByRole('button', { name: /Choose a dashboard/ }))
+        userEvent.click(chooseDashboard)
+
+        const dashboard2 = screen.getByRole('option', { name: /Global Dashboard 2/ })
+        userEvent.click(dashboard2)
+
+        expect(history.location.pathname).toEqual('/insights/dashboards/bar')
+    })
+
+    // Note: the rest of these are unwritten due to a bug in ReachUI.
```
Comment (vovakulikov) [inline client/web/src/enterprise/insights/pages/dashboards/dashboard-page/components/dashboards-content/DashboardsContent.test.tsx:168]: Spent a lot of time trying to find the cause of this problem. My guess is that @reach/descendants operate with jsDOM too heavily. This package is needed to manipulate with collections of the child components. The root component (in our case Menu or DropdownProvider) is responsible for calling right onSelect but because jestDom doesn't work exactly like JSDOM we have a problem with that. But it's a long shot maybe the problem is somewhere else. 

An interesting thing is that there is an issue in the reach UI repo https://github.com/reach/reach-ui/issues/833 So callback can be called somehow.

**Review by vovakulikov — APPROVED** [review]
LGTM. It's a bummer that reach ui has this problem. @unclejustin do you think we should cover the rest cases in puppeteer tests for now? (or do you have a plan to investigate this problem with reach ui more?)

File: client/web/src/enterprise/insights/pages/dashboards/dashboard-page/components/dashboards-content/DashboardsContent.test.tsx (hunk 2 above)
Comment (unclejustin) [reply client/web/src/enterprise/insights/pages/dashboards/dashboard-page/components/dashboards-content/DashboardsContent.test.tsx:168]: Honestly, I think Reach is wrong to use `onSelect` at all. It is my understanding that `onSelect` is reserved for text inputs. There is a slight workaround we can use where we switch to `onClick` instead which I believe is the correct usage here. 

"Won't that break keyboard navigation?" I am unsure. But considering that Reach is relying on some weird loophole for this functionality right now, I'm not sure that its current implementation is better.

tldr adding onClick to the MenuItem also works and enables the tests. I just didn't want to change the production code to accommodate a test.

**Comment by unclejustin** [comment]
@vovakulikov re: expanding tests

I'm going to push Reach to address the issue I opened. I'd like to give them a chance to respond before we decide.
